pkg/initpkg/context.go

```diff
@@ -0,0 +1,277 @@
+package initpkg
+
+import (
+	"crypto/sha512"
+	"fmt"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+
+	// apko_types "chainguard.dev/apko/pkg/build/types"
+	melange_build "chainguard.dev/melange/pkg/build"
+	melange_util "chainguard.dev/melange/pkg/util"
+	yam "github.com/chainguard-dev/yam/pkg/yam/formatted"
+	"github.com/dprotaso/go-yit"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v3"
+)
+
+type Context struct {
+	URI     string
+	Name    string
+	Version string
+	License string
+	Layout  string
+	WorkDir string
```

Comment:
Consider having `Context` taking a `WorkFS` parameter so it's more easily tested with an `apkfs.FullFS` or something instead of always writing to disk.